### PR TITLE
send results via gRPC

### PR DESCRIPTION
**Pull request by avobl**

### CODE REVIEW

File: internal/checker/checker.go [hunk 1]
```diff
@@ -64,44 +68,88 @@ func (c *checker) Start() {
...
-		// set it to nil for stop processing previous result while no messages from Kafka.
+		results = append(results, res...)
+	}
+	totalResult.Result = append(totalResult.Result, results...)
+	return &totalResult
+}
+
+// sendResult sends given results via gRPC.
+func (c *checker) sendResult(r *result.Result) error {
+	conn, err := grpc.Dial(c.config.GRPCConfig.ResultCollectorAddress, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
```
Comment (igkostyuk) [inline internal/checker/checker.go]: Do we really need to open a connection every time we send? You can add a client to the checker struct

File: internal/checker/checker.go [hunk 2]
```diff
@@ -64,44 +68,88 @@ func (c *checker) Start() {
...
 }
 
 // ErrorBasedCheck firstly reads payloads from the file, then creates regexp for errors, after gets site body
 // checking it on possible inclusions of error keywords, finally posts form on site with injectable parameters.
 // Having this done matches response body with possible error keywords.
-func (c *checker) ErrorBasedCheck(link string) error {
+func (c *checker) ErrorBasedCheck(link string) ([]string, error) {
+	results := make([]string, 0)
 	// reading file with payloads with path from config.
 	f, err := os.Open(c.config.Checker.ErrBasedPayload)
```
Comment (igkostyuk) [inline internal/checker/checker.go:123]: I think reading the file on each run is not a really good idea. You can move it to main.

File: internal/checker/checker.go [hunk 3]
```diff
@@ -64,44 +68,88 @@ func (c *checker) Start() {
...
 	var wg sync.WaitGroup
 
 	scan := bufio.NewScanner(f)
 	for scan.Scan() {
 		payload := scan.Text()
 		wg.Add(1)
-		go c.submitForm(link, payload, countBefore, forms, &wg)
+		go func() {
+			res := c.submitForm(link, payload, countBefore, forms, &wg)
+			results = append(results, res...)
```
Comment (igkostyuk) [inline internal/checker/checker.go]: Isn't it race condition?

Comment (avobl) [reply internal/checker/checker.go]: yep, you're absolutely right! I'll fix it. Thank you.

File: internal/checker/checker.go (hunk 2 above)
Comment (avobl) [reply internal/checker/checker.go:123]: There are a lot of payloads in the file. I'm not sure about hold then in memory all the time. However it's a really nice question which approach is more suitable here.

**Pull request merged**